### ISSUE TITLE
2.0.3: gems with directory names in data.tar.gz fail to install

### ISSUE THREAD

**Issue by marksolaris** [rubygems/rubygems#631]
Solaris 10 & 11, ruby 2.0.0p247 with gem 2.0.3

If I patch a gem and then rebuild it, having the directory names in the data.tar.gz file will result in a 'file exists' error.

```
tar -xf /usr/local/src/ruby/orig/ffi-1.9.0.gem
mkdir data; cd data
untgz ../data.tar.gz
patch -p0 < /usr/local/src/ruby/ffi-1.9.0.gem.diff
tar -cf - . | gzip -9c > ../data.tar.gz
cd ..
tar -cf /usr/local/src/ruby/ffi-1.9.0.gem data.tar.gz metadata.gz
```

The above commands create a data.tar.gz with the '.' as the leading char. That causes:

```
ERROR:  While executing gem ... (Errno::EEXIST)
    File exists - /usr/local/lib/ruby/gems/2.0.0/gems/ffi-1.9.0
```

This file is created:

```
host:/usr/local/lib/ruby/gems/2.0.0/gems root# ls -alCF /usr/local/lib/ruby/gems/2.0.0/gems/ffi*
-rwxr-xr-x   1 root     root           0 Sep  2 14:14 ffi-1.9.0*
```

If I use the command

```
tar -cf - * | gzip -9c > ../data.tar.gz
```

then I get

```
ERROR:  While executing gem ... (Errno::EEXIST)
    File exists - /usr/local/lib/ruby/gems/2.0.0/gems/ffi-1.9.0/ext
```

and this file is created:

```
host:/ root# ls -alCF /usr/local/lib/ruby/gems/2.0.0/gems/ffi-1.9.0
-rwxr-xr-x   1 root     root           0 Sep  2 13:59 ext*
```

I have to use the command

```
tar -cf - `find * -type f -print` | gzip -9c > ../data.tar.gz
```

to just list the filenames, and not any directory names, in the data.tar.gz

In other words any directory that exists (first?) in the data.tar.gz ends up becoming a zero byte file and causing the install to break.
